### PR TITLE
Add TextureSize test

### PR DESCRIPTION
**Pull request by qjia7**
This test exposes two issues:

1. On NVIDIA drivers, the HLSL Texture.GetDimensions instruction doesn't implicitly add the Texture's base level to the lod parameter. This is inconsistent with the behaviour of the reference rasterizer.

2. When the HLSL shader calls GetDimensions on a Texture whose backing SRV only specifies a limited number of mips, NVIDIA drivers can query properties of the whole texture, while the reference rasterizer returns default values for mips that are out of range of the SRV.

### CODE REVIEW

**Comment by qjia7** [comment]
@null77 Please take a look.

File: TextureSize/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+In Intel driver, It don't need to add the offset in GetDimensions.
+
+### Correct rendering
+
+![correct rendering](correct.PNG?raw=true)
+
+### Incorrect rendering: Intel
+
+![incorrect rendering](incorrect.PNG?raw=true)
+
```
Comment (null77) [inline TextureSize/README.md]: I think you might have this backwards -- Intel and AMD have the correct behaviour and NVIDIA has a bug. You might want to swap correct/incorrect, and name the incorrect.PNG incorrect-nvidia.PNG. Also if you could add a section documenting which good and bad driver versions you tested, as in the other samples, that'd be appreciated.

File: TextureSize/Tutorial03.fx [hunk 2]
```diff
@@ -0,0 +1,53 @@
+//--------------------------------------------------------------------------------------
+// Vertex Shader
+//--------------------------------------------------------------------------------------
+
+// texSizeCase = {lod, baseLevel, expectedWith, expectedHeight}
+uniform int4 texSizeCase : register(b0);
+
+static const uint u_sampler = 0;
```
Comment (null77) [inline TextureSize/Tutorial03.fx]: This is very good, but I think you could simplify this a little bit by taking out the ANGLE specific stuff. Not really necessary, it's OK the way it is, but if you feel like cleaning it up feel free. (Again, not necessary)

**Comment by null77** [comment]
This looks very good. If you could fix the incorrect/correct ordering we can land this and potentially share this with NVIDIA/MS so they can fix their behaviour.

File: TextureSize/README.md (hunk 1 above)
Comment (qjia7) [inline TextureSize/README.md]: @null77 I have updated the description. But I don't have a NVIDIA's gpu. So I labeled NVIDIA as 'Didn't try'. Please help update this field when this PR is merged.

File: TextureSize/Tutorial03.fx (hunk 2 above)
Comment (qjia7) [inline TextureSize/Tutorial03.fx]: Done. Thanks.

**Comment by qjia7** [comment]
@null77 Please take another look.

File: TextureSize/README.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# Texture Size issue
+
+This test exposes two issues:
+1. NVIDIA has bug in GetDimensions when base level is greater than zero.
```
Comment (null77) [inline TextureSize/README.md]: I think we need to be more explicit. Something like:

"On NVIDIA drivers, the HLSL Texture.GetDimensions instruction doesn't implicitly add the Texture's base level to the lod parameter. This is inconsistent with the behaviour of the reference rasterizer."

File: TextureSize/README.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+The workaround is to add the baseLevel in GetDimensions.
+
+2. Open: Need D3D spec to clearly point it out that whether the shader can access the mipLevel
```
Comment (null77) [inline TextureSize/README.md]: "When the HLSL shader calls GetDimensions on a Texture whose backing SRV only specifies a limited number of mips, NVIDIA drivers can query properties of the whole texture, while the reference rasterizer returns default values for mips that are out of range of the SRV."

File: TextureSize/README.md [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+info when the mipLevel is out of ShaderResourceView's MipLevels - 1?
+Comment line 'int mipLevels = lod + baseLevel + 1;' and use 'int mipLevels = lod + baseLevel;'.
+In Intel driver, it will fail since it can't access the level info that is out of SRV's levels.
+
+### Correct rendering
+
+![correct rendering](correct.PNG?raw=true)
+
+### Incorrect rendering:
```
Comment (null77) [inline TextureSize/README.md]: Incorrect rendering (NVIDIA):

File: TextureSize/README.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+
+![incorrect rendering](incorrect.PNG?raw=true)
```
Comment (null77) [inline TextureSize/README.md]: incorrect-nvidia.PNG

File: TextureSize/README.md [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+### Tested driver versions:
+
+* GOOD: Intel 20.19.15.4463 (5-25-2016)
+* BAD: NVIDIA (Didn't try. It should fail therotically.)
```
Comment (null77) [inline TextureSize/README.md]: BAD: NVIDIA 10.18.13.6881 (7-10-2016)

Just verified the output is as expected.

**Comment by null77** [comment]
Thanks, just a few minor fixes, if you don't mind, and I'll land this.

**Comment by qjia7** [comment]
@null77 Have solved all your comments. PTAL. Thanks.

**Comment by null77** [comment]
Looks great, thanks!

**Pull request merged**